connection/connection.go

```diff
@@ -130,18 +132,68 @@ func NewConnection(u url.URL,
 		writeWait:       writeWait,
 		maxMessageSize:  maxMessageSize,
 		defaultPingWait: defaultPingWait,
+		done:            make(chan bool),
 	}
-	// ping-pong
 	ws.SetReadLimit(maxMessageSize)
-	ws.SetReadDeadline(time.Now().Add(defaultPingWait))
-	ws.SetPingHandler(func(message string) error {
-		pongWait, _ := strconv.Atoi(message)
-		ws.SetReadDeadline(time.Now().Add(time.Duration(pongWait) * time.Second))
+
+	go c.pingPongHandler()
+
+	return c, nil
+}
+
+func (c *Connection) pingPongHandler() {
+	// handle the ping-pong connection health check
+	err := c.connection.SetReadDeadline(time.Now().Add(c.defaultPingWait))
+	if err != nil {
+		return
+	}
+
+	pingPeriod := (c.defaultPingWait * 9) / 10
+	ticker := time.NewTicker(pingPeriod)
+	defer ticker.Stop()
+
+	c.connection.SetPongHandler(func(string) error {
+		log.Debug("PongHandler called")
+		// requires go >= 1.15
+		// ticker.Reset(pingPeriod)
```

Comment:
golang 1.14 is the latest supported in yocto warrior (and we did the port of this version ourselves). With the current plan, we won't be able to use 1.15 features until April 2023.

Reply:
This is not an issue, it's a (micro) optimisation and that's why it is commented.